fwdctl/src/dump.rs

```diff
@@ -24,50 +28,230 @@ pub struct Options {
     /// Defaults to None.
     #[arg(
         short = 's',
-        long = "start-key",
+        long,
         required = false,
         value_name = "START_KEY",
         value_parser = key_parser,
         help = "Start dumping from this key (inclusive)."
     )]
     pub start_key: Option<Key>,
+
+    /// The key to stop dumping to (if no key is provided, stop to the end).
+    /// Defaults to None.
+    #[arg(
+        short = 'S',
+        long,
+        required = false,
+        value_name = "STOP_KEY",
+        value_parser = key_parser,
+        help = "Stop dumping to this key (inclusive)."
+    )]
+    pub stop_key: Option<Key>,
+
+    /// The max number of the keys going to be dumped.
+    /// Defaults to None.
+    #[arg(
+        short = 'm',
+        long,
+        required = false,
+        value_name = "MAX_KEY_COUNT",
+        help = "Maximum number of keys going to be dumped."
+    )]
+    pub max_key_count: Option<u32>,
+
+    /// The output format of database dump.
+    /// Possible Values: ["csv", "json", "stdout"].
+    /// Defaults to "stdout"
+    #[arg(
+        short = 'o',
+        long,
+        required = false,
+        value_name = "OUTPUT_FORMAT",
+        value_parser = ["csv", "json", "stdout"],
+        default_value = "stdout",
+        help = "Output format of database dump, default to stdout. CSV and JSON formats are available."
+    )]
+    pub output_format: String,
```

Comment:
Can you use an enum here and let clap handle the errors?

Reply:
I think when declaring `value_parser` in clap, it would work just like using an Enum.